File: app/models/oauth_access_token.rb
# typed: strict
class OauthAccessToken < ApplicationRecord
  # Associations
  belongs_to :user, foreign_key: :user_id
  belongs_to :oauth_application, foreign_key: :application_id

  # Validations
  validates :resource_owner_id, presence: true
  validates :token, presence: true, uniqueness: true
  validates :refresh_token, presence: true, uniqueness: true
  validates :expires_in, presence: true
  validates :scopes, presence: true
  validates :resource_owner_type, presence: true
  validates :refresh_expires_in, presence: true
end

File: src/models/OauthAccessToken.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, JoinColumn } from 'typeorm';
import { OauthApplication } from './OauthApplication';

@Entity('oauth_access_tokens')
export class OauthAccessToken {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ type: 'timestamp', default: () => 'CURRENT_TIMESTAMP' })
  created_at: Date;

  @Column({ type: 'timestamp', default: () => 'CURRENT_TIMESTAMP' })
  updated_at: Date;

  @Column()
  resource_owner_id: string;

  @Column()
  token: string;

  @Column({ nullable: true })
  refresh_token: string;

  @Column('int')
  expires_in: number;

  @Column({ type: 'timestamp', nullable: true })
  revoked_at: Date;

  @Column('text', { array: true, default: '{}' })
  scopes: string[];

  @Column({ nullable: true })
  previous_refresh_token: string;

  @Column()
  resource_owner_type: string;

  @Column('int', { nullable: true })
  refresh_expires_in: number;

  @Column()
  application_id: string;

  @ManyToOne(() => OauthApplication)
  @JoinColumn({ name: 'application_id' })
  application: OauthApplication;
}